packages/dev/loaders/src/SPLAT/splatFileLoader.ts

```diff
@@ -79,49 +103,290 @@ export class SPLATFileLoader implements ISceneLoaderPluginAsync, ISceneLoaderPlu
      * @returns a promise containing the loaded meshes, particles, skeletons and animations
      */
     public async importMeshAsync(
-        _meshesNames: any,
+        meshesNames: any,
         scene: Scene,
         data: any,
         rootUrl: string,
         onProgress?: (event: ISceneLoaderProgressEvent) => void,
         fileName?: string
     ): Promise<ISceneLoaderAsyncResult> {
-        const gaussianSplatting = new GaussianSplattingMesh("GaussianSplatting", null, scene);
-        await gaussianSplatting.loadFileAsync(rootUrl + (fileName ?? ""));
-        return {
-            meshes: [gaussianSplatting],
-            particleSystems: [],
-            skeletons: [],
-            animationGroups: [],
-            transformNodes: [],
-            geometries: [],
-            lights: [],
-            spriteManagers: [],
+        return this._parse(meshesNames, scene, data, rootUrl).then((meshes) => {
+            return {
+                meshes: meshes,
+                particleSystems: [],
+                skeletons: [],
+                animationGroups: [],
+                transformNodes: [],
+                geometries: [],
+                lights: [],
+                spriteManagers: [],
+            };
+        });
+    }
+
+    private _buildPointCloud(pointcloud: PointsCloudSystem, data: ArrayBuffer): boolean {
+        if (!data.byteLength) {
+            return false;
+        }
+        const uBuffer = new Uint8Array(data);
+        const fBuffer = new Float32Array(uBuffer.buffer);

```

Comment:
Isn't this the same as `new Float32Array(data)`? Assuming yes, that would have been more clear to me.